torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -958,7 +965,7 @@ std::shared_ptr<JitFuture> TensorPipeAgent::send(
 
         pipeRead(
             clientPipe.pipe_,
-            [this, &clientPipe, expirationTime](
+            [this, &clientPipe, messageIdR = messageId](
```

Comment:
So IIUC, this means that it's not necessary that `pipeRead` here is guaranteed to read the response that was sent in the associated `pipeWrite` above?

Reply:
Yes, I think this is by design of tensorpipe. I think the pipeRead is just called to ensure we fire a "read" request to process an incoming response, but it doesn't necessarily have to be the response for the same request. It will just read whatever it receives on the wire.